From 42a1e30ef4c42bb9a16510476a42cf0349c2aee1 Mon Sep 17 00:00:00 2001
Message-Id: <42a1e30ef4c42bb9a16510476a42cf0349c2aee1.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Mon, 14 Jul 2014 12:43:44 +0530
Subject: [PATCH 165/312] REVERTME [VPG]: drm/i915: Fix the switch name for
 VLV HDMI.

This patch fixes the switch name for VLV HDMI
as required by HAD driver. This patch is a REVERTME as
HDMI audio design might be changing later.

Change-Id: I472ac982394bd36e8582b7572684a12d5664e37a
Signed-off-by: johan.bohlin <johan.bohlin@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_hdmi.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index b0c6875..19ac8a5 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -1636,6 +1636,10 @@ void intel_hdmi_init_connector(struct intel_digital_port *intel_dig_port,
 	if (IS_ENABLED(CONFIG_SWITCH)) {
 		intel_connector->hotplug_switch.name =
 			kasprintf(GFP_KERNEL, "hdmi_%c", 'a' + port);
+#ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
+		if (IS_VALLEYVIEW(dev))
+			intel_connector->hotplug_switch.name = "hdmi";
+#endif
 		if (!intel_connector->hotplug_switch.name) {
 			DRM_ERROR("%s failed to allocate memory", __func__);
 			kfree(intel_connector);
-- 
1.7.9.5

